docs/docsite/rst/playbook_guide/playbooks_reuse_roles.rst

```diff
@@ -156,7 +156,7 @@ When using ``vars:`` within the ``roles:`` section of a playbook, the variables
 Including roles: dynamic reuse
 ------------------------------
 
-You can reuse roles dynamically anywhere in the ``tasks`` section of a play using ``include_role``. While roles added in a ``roles`` section run before any other tasks in a play, included roles run in the order they are defined. If there are other tasks before an ``include_role`` task, the other tasks will run first.
+You can reuse roles dynamically anywhere in the ``tasks`` section of a play using ``include_role``. While roles added in a ``roles`` section run before any other tasks in a play, included roles run in the order they are defined. If there are other tasks before an ``include_role`` task, the other tasks will run first. Ansible does not deduplicate included roles, even if they have the exact same parameters, and setting ``allow_duplicates: false`` has no effect on ``include_role``.
```

Comment:
It does but it is a bit limited, it includes the vars context for the 'signature' to dedupe on, which makes it very hard for the `include_role` signatures to match. `import_role`, being run in a 'preprocess' state don't have the variability present so the signature is a lot more stable, matching the behvior of `roles:`.